phoenix-core/src/main/java/org/apache/phoenix/mapreduce/OrphanViewTool.java

```diff
@@ -0,0 +1,812 @@
+    private static final Logger LOG = LoggerFactory.getLogger(OrphanViewTool.class);
+    // Query all the views that are not "MAPPED" views
+    private static final String viewQuery = "SELECT " +
+            TENANT_ID + ", " +
+            TABLE_SCHEM + "," +
+            TABLE_NAME +
+            " FROM " + SYSTEM_CATALOG_NAME +
+            " WHERE "+ TABLE_TYPE + " = '" + PTableType.VIEW.getSerializedValue() +"' AND NOT " +
+            VIEW_TYPE + " = " + PTable.ViewType.MAPPED.getSerializedValue();
+    // Query all physical links
+    private static final String physicalLinkQuery = "SELECT " +
+            TENANT_ID + ", " +
+            TABLE_SCHEM + ", " +
+            TABLE_NAME + ", " +
+            COLUMN_NAME + " AS PHYSICAL_TABLE_TENANT_ID, " +
+            COLUMN_FAMILY + " AS PHYSICAL_TABLE_FULL_NAME " +
+            " FROM " + SYSTEM_CATALOG_NAME +
+            " WHERE "+ LINK_TYPE + " = " +
+            PTable.LinkType.PHYSICAL_TABLE.getSerializedValue();
+    // Query all child-parent links
+    private static final String childParentLinkQuery = "SELECT " +
+            TENANT_ID + ", " +
+            TABLE_SCHEM + ", " +
+            TABLE_NAME + ", " +
+            COLUMN_NAME + " AS PARENT_VIEW_TENANT_ID, " +
+            COLUMN_FAMILY + " AS PARENT_VIEW_FULL_NAME " +
+            " FROM " + SYSTEM_CATALOG_NAME +
+            " WHERE "+ LINK_TYPE + " = " +
+            PTable.LinkType.PARENT_TABLE.getSerializedValue();
+    // Query all parent-child links
+    private static final String parentChildLinkQuery = "SELECT " +
+            TENANT_ID + ", " +
+            TABLE_SCHEM + ", " +
+            TABLE_NAME + ", " +
+            COLUMN_NAME + " AS CHILD_VIEW_TENANT_ID, " +
+            COLUMN_FAMILY + " AS CHILD_VIEW_FULL_NAME " +
+            " FROM " + SYSTEM_CHILD_LINK_NAME +
+            " WHERE "+ LINK_TYPE + " = " +
+            PTable.LinkType.CHILD_TABLE.getSerializedValue();
+
+    // Query all the tables that can be a base table
+    private static final String candidateBaseTableQuery = "SELECT " +
+            TENANT_ID + ", " +
+            TABLE_SCHEM + ", " +
+            TABLE_NAME +
+            " FROM " + SYSTEM_CATALOG_NAME +
+            " WHERE "+ " NOT " + TABLE_TYPE + " = '" + PTableType.VIEW.getSerializedValue() + "'";
+
+    private String outputPath;
+    private String inputPath;
+    private boolean clean = false;
+    private int maxViewLevel = 0;
+    private static final long defaultAge = 24*60*60*1000; // 1 day
+    private long age = 0;
+
+    private static final byte VIEW = 0;
+    private static final byte PHYSICAL_TABLE_LINK = 1;
+    private static final byte PARENT_TABLE_LINK = 2;
+    private static final byte CHILD_TABLE_LINK = 3;
+    private static final byte ORPHAN_TYPE_COUNT = 4;
+
+    BufferedWriter writer[] = new BufferedWriter[ORPHAN_TYPE_COUNT];
+    BufferedReader reader[] = new BufferedReader[ORPHAN_TYPE_COUNT];
+    HashMap<Key, View> orphanViewSet = new HashMap<>();
```

Comment:
Can you add some comments here about what is the data organization i.e. what is the expected key, value, list of what, etc.

Reply:
ok